src/FastSerialization/StreamReaderWriter.cs

```diff
@@ -710,6 +711,18 @@ public override void Read(byte[] data, int offset, int length)
             // remainder from the stream
             if (length > (bytes.Length - align))
             {
+                // check if our inputstreambytesread < position in stream then throw out delta
+                if (inputStreamBytesRead < positionInStream)
+                {
+                    long amountToThrowOut = positionInStream - inputStreamBytesRead;
+                    while (amountToThrowOut > 0)
+                    {
+                        int read = inputStream.Read(bytes, 0, (int)Math.Min(amountToThrowOut, (long)bytes.Length));
```

Comment:
nit: Did you need to handle Read() returning 0?

Reply:
I think we do. That would signify we've read past the end of the stream. In the live scenario, it would block until the stream closed in which case, we would also need to error if we read 0 bytes. I think there's an exception defined for that I can add here.

Reply:
Doesn't read returning 0 signify end of stream? The error is a bit unclear to me as it stands.